### PR TITLE
Rename enable_secure_identifiers -> enforce_strict_identifier_format

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

Another issue with this setting that once it's enabled by default, server cannot start (e.g. some system tables have column `ProfileEvents.Names`)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 2d58d7d19ed8444025e056a91556c65544396151 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70335/2d58d7d19ed8444025e056a91556c65544396151/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70335/2d58d7d19ed8444025e056a91556c65544396151/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/executeQuery.cpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ namespace Setting
     extern const SettingsBool use_query_cache;
     extern const SettingsBool wait_for_async_insert;
     extern const SettingsSeconds wait_for_async_insert_timeout;
-    extern const SettingsBool enable_secure_identifiers;
+    extern const SettingsBool mongodb_throw_on_unsupported_query;
```
Comment (yariks5s) [inline src/Interpreters/executeQuery.cpp:157]: I guess the wrong setting name here

**Review by yariks5s — APPROVED** [review]

File: src/Core/Settings.cpp [hunk 2]
```diff
@@ -5497,7 +5497,7 @@ Replace external dictionary sources to Null on restore. Useful for testing purpo
     M(Bool, create_if_not_exists, false, R"(
 Enable `IF NOT EXISTS` for `CREATE` statement by default. If either this setting or `IF NOT EXISTS` is specified and a table with the provided name already exists, no exception will be thrown.
 )", 0) \
-    M(Bool, enable_secure_identifiers, false, R"(
+    M(Bool, enforce_strict_identifier_format, false, R"(
 If enabled, only allow secure identifiers which contain only underscore and alphanumeric characters
```
Comment (vdimir) [inline src/Core/Settings.cpp:5501]:
```suggestion
If enabled, only allow identifiers containing alphanumeric characters and underscores.
```

**Comment by clickgapai** [comment]
Hi @vdimir @yariks5s — while reviewing this PR I found the following:

- **Bug (P2):** `enforce_strict_identifier_format` does not validate constraint names due to `ASTConstraintDeclaration` bypassing `writeIdentifier` — https://github.com/ClickHouse/ClickHouse/issues/104630

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**